src/History/DragAndDrop.jsx

```diff
@@ -0,0 +1,50 @@
+import { useRef, useState } from "react";
+
+import KeywordGroup from "./KeywordGroup";
+
+export default function DragAndDrop() {
+  const dragPosition = useRef();
+  const [list, setList] = useState([
+    { id: 1, name: "Keyword 1", keywords: [] },
+    { id: 2, name: "Keyword 2", keywords: ["keyword1", "keyword2"] },
+    { id: 3, name: "Keyword 3", keywords: ["keyword3"] },
+    { id: 4, name: "Keyword 4", keywords: ["keyword4", "keyword5"] },
+    {
+      id: 5,
+      name: "Keyword 4",
+      keywords: ["keyword6", "keyword7", "keyword8"],
+    },
+  ]);
+
+  const dragStart = (event, groupIndex, keyword) => {
+    dragPosition.current = { groupIndex, keyword };
+  };
+
+  const drop = (event, groupIndex) => {
+    event.preventDefault();
+    const newList = [...list];
+    const { groupIndex: dragGroupIndex, keyword } = dragPosition.current;
+
+    newList[dragGroupIndex].keywords = newList[dragGroupIndex].keywords.filter(
+      (notKeyword) => notKeyword !== keyword
+    );
+    newList[groupIndex].keywords.push(keyword);
+
+    dragPosition.current = null;
+    setList(newList);
+  };
+
+  return (
+    <>
+      {list.map((item, index) => (
+        <KeywordGroup
+          key={item.id}
+          keyword={item.name}
+          keywords={item.keywords}
```

Comment:
keyword가 group 명을 표현하고 있는 것 같아서, 살짝 다른 단어를 사용하는 것은 어떨까요? 목업 작업할 때는 keyword라고 굵은 글씨로 쓰여있어서 무엇을 뜻하는 지 바로 알 수 있었는데, 코드상으로만 보니까 어려운 것 같아서 조심스럽게 의견내 보았습니다 :)

```suggestion
          groupName={item.name}
          keywords={item.keywords}
```

Reply:
앗 의견주셔서 감사합니다! 기존에 Git Conflict 된 사항들 수정한 단계라 혹시나 merge가 안될 상황에 대비하여 다음 브랜치 기능 구현시에 잊지않고 변수명 변경하여  작업해두겠습니다! :)

Reply:
오 그러면 빠른 진행을 위해 바로 승인하겠습니다